pkg/datapath/policy/policymap.go

```diff
@@ -0,0 +1,273 @@
+// Copyright 2016-2019 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package policy
+
+import (
+	"fmt"
+
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/maps/policymap"
+	"github.com/cilium/cilium/pkg/policy"
+)
+
+var (
+	log = logging.DefaultLogger.WithField(logfields.LogSubsys, "datapath-policymap")
+)
+
+type PolicyMapImplementer struct {
+	policymap *policymap.PolicyMap
+	id        uint16
+	isInit    bool
+}
+
+func (p *PolicyMapImplementer) IsInit() bool {
+	return p.isInit
+}
+
+func (p *PolicyMapImplementer) GetFd() int {
```

Comment:
Composition of the underlying map would allow skipping declaration of this kind of boilerplate.